packages/cli/src/ActiveWorkflowRunner.ts

```diff
@@ -99,7 +99,11 @@ export class ActiveWorkflowRunner implements IWebhookManager {
 	) {}
 
 	async init() {
-		if (config.getEnv('executions.mode') === 'queue' && config.getEnv('multiMainSetup.enabled')) {
+		if (
+			config.getEnv('executions.mode') === 'queue' &&
+			config.getEnv('multiMainSetup.enabled') &&
+			config.getEnv('generic.instanceType') === 'main'
+		) {
```

Comment:
Maybe extract this check to a helper such as `shouldEnrollInMultiMainElection`? Just because we're duplicating this code a lot.

Reply:
An alternative would be to move this check to inside the `init` method.